src/index/CompressedRelation.cpp

```diff
@@ -102,8 +102,10 @@ CompressedRelationReader::asyncParallelBlockGenerator(
     CompressedBlock compressedBlock =
         readCompressedBlockFromFile(blockMetadata, columnIndices);
     lock.unlock();
+    cancellationHandle->throwIfCancelled();
     auto decompressedBlockAndMetadata = decompressAndPostprocessBlock(
         compressedBlock, blockMetadata.numRows_, scanConfig, blockMetadata);
+    cancellationHandle->throwIfCancelled();
```

Comment:
These changes probably don't hurt, but I don't understand what they are supposed to fix.
In theory, these lead to poor performance, as the cancellationHandle is concurrently accessed by many threads, and so we have added 2 L2-Cache-misses, but probably even those are not fixable.

The issues we saw, were reported here, but the missed cancellation check was somewhere else (your meachnism doesn't report, where the cancellation checks were missing, but the first check after the missed deadline. With materialized inputs that is often in the same loop, but with lazy inputs, it might be completely somewhere else, in a different generator. 
So I don't know, what the benefit of these checks is (reading one block should definitely be orders of magnituds faster than the timeout interval, so it suffices if the consumer checks the timeout.